### ISSUE TITLE
How to allow for preprocessing of image channel values in ImageDataGenerator and flow_from_directory?

### ISSUE THREAD

**Issue by pchankh** [fchollet/keras#4630]
In short we only see rescale option in ImageDataGenerator.  How do you do the following for pertained VGG16 network that needs the channels values preprocessed as follows:
    im = cv2.resize(cv2.imread('cat.jpg'), (224, 224)).astype(np.float32)
    im[:,:,0] -= 103.939
    im[:,:,1] -= 116.779
    im[:,:,2] -= 123.68
    im = im.transpose((2,0,1))
    im = np.expand_dims(im, axis=0)
Where do we specify these options in ImageDataGenerator?

Thanks
Dr

**Comment by mabirck** [comment]
There is a tread #3338 and  gives a solution by adding a pipeline with fork of imageGenerator in keras. I think is that your looking for, sorry if I am wrong.
